.github/pull_request_template.md

```diff
@@ -1,15 +1,67 @@
+<!--
+    Please review the contributing guide before submitting: https://developers.home-assistant.io/docs/android/submit
+    Please, complete the following sections to help the processing and review of your changes.
+    Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
 
-<!-- Thank you for submitting a Pull Request and helping to improve Home Assistant. Please complete the following sections to help the processing and review of your changes. Please do not delete anything from this template. -->
+    Thank you for submitting a Pull Request and helping to improve Home Assistant. You are amazing!
+-->
 
 ## Summary
-<!-- Provide a brief summary of the changes you have made and most importantly what they aim to achieve -->
+<!--
+    Provide a brief summary of the changes you have made and most importantly what they aim to achieve.
+    Don't forget any links that could be useful to the reader. (Github issues, PRs, documentation, articles, ...)
+
+    * What was the motivation behind this change?
+    * What is the impact of the changes on the application?
+-->
+
+## Checklist
+<!--
+    Put an `x` in the boxes that apply. You can also fill these out after
+    creating the PR. If you're unsure about any of them, don't hesitate to ask.
+    We're here to help! This is simply a reminder of what we are going to look
+    for before merging your code.
+-->
+
+- [ ] New or updated tests have been added to cover the changes following the testing [guidelines](https://developers.home-assistant.io/docs/android/testing/introduction).
+- [ ] The code follows the project's [code style](https://developers.home-assistant.io/docs/android/codestyle) and [best_practices].(https://developers.home-assistant.io/docs/android/best_practices).
+- [ ] The changes have been thoroughly tested, and edge cases have been considered.
+- [ ] The changes are backward compatible or any compatibility break is documented with previous versions of Home Assistant Core.
 
 ## Screenshots
-<!-- If this is a user-facing change not in the frontend, please include screenshots in light and dark mode. -->
+<!--
+    If this is a user-facing change not in the frontend, please include screenshots in light and dark mode.
+
+    Note: Remove this section if there are no screenshots.
+-->
+
+## Link to pull request in documentation repositories
+<!-- 
+    This pull request introduces, changes, or removes user-facing functionality.
```

Comment:
I wonder if we can merge this comment block and the next one, and end with two places to link documentation, similar to how the Home Assistant Core PR template has one big section to link to other PRs (like docs). They are both about docs and the text is 90% the same. What do you think?

Reply:
I though about it while writing it. I took this approach thinking it would be more explicit for ppl that it might need 2 PRs.
I'm not fully convince about this but since the comments is hidden I think we could try like this for a while.